mlrun/datastore/utils.py

```
@@ -222,3 +222,51 @@ def validate_additional_filters(additional_filters):
             for sub_value in value:
                 if isinstance(sub_value, float) and math.isnan(sub_value):
                     raise mlrun.errors.MLRunInvalidArgumentError(nan_error_message)
+
+
+class KafkaParameters:
+    def __init__(self, kwargs: dict):
+        import kafka
+
+        self._kafka = kafka
+        self._kwargs = kwargs
+        self._client_configs = {
+            "consumer": self._kafka.KafkaConsumer.DEFAULT_CONFIG,
+            "producer": self._kafka.KafkaProducer.DEFAULT_CONFIG,
+            "admin": self._kafka.KafkaAdminClient.DEFAULT_CONFIG,
+        }
+
+    def _get_config(self, client_type: str) -> dict:
+        res = {
+            k: self._kwargs[k]
+            for k in self._kwargs.keys() & self._client_configs[client_type].keys()
```

I don't like this key intersection, as it doesn't validate anything about `self._kwargs`.
For example - a typo in the passed kwargs will silently fail here.

---

Hmm, good point. Let me clarify the attributes dictionary structure:

The attributes dictionary contains three categories:
1. Internal attributes used by KafkaOutputStream (e.g., "brokers", "topics", "sasl")
2. Common parameters passed to all Kafka clients (KafkaConsumer, KafkaProducer, KafkaAdminClient) such as 'ssl_check_hostname'
3. Client-specific parameters like 'metrics_sample_window_ms' which only apply to certain clients (e.g., KafkaConsumer)

All these different attribute types are currently maintained in a single dictionary. I'll implement a verification mechanism that validates the attribute list against the complete set of possible keys from all these categories.